knora/dsplib/utils/onto_create_ontology.py

```diff
@@ -13,598 +13,618 @@
+                method="create",
+                error_message_on_failure=f"\tWARNING: Unable to create group '{group_name}'."
+            )
         except BaseError as err:
-            print(f"ERROR while trying to create group '{group_name}'. The error message was: {err.message}")
-            exit(1)
-        except Exception as exception:
-            print(f"ERROR while trying to create group '{group_name}'. The error message was: {exception}")
-            exit(1)
-    return new_groups
+            print(err.message)
+            overall_success = False
+            continue
+
+        current_project_groups[group_remote.name] = group_remote
+        print(f"\tCreated group '{group_name}'.")
 
+    return current_project_groups, overall_success
 
-def create_users(con: Connection, users: list[dict[str, str]], groups: dict[str, Group], project: Project,
-                 verbose: bool) -> None:
+
+def _create_users(
+    con: Connection,
+    users: list[dict[str, str]],
+    current_project_groups: dict[str, Group],
+    current_project: Project,
+    verbose: bool
+) -> bool:
     """
-    Creates user(s) on a DSP server from a list of user definitions
+    Creates users on a DSP server from the "users" section of a JSON project file. If a user cannot be created, a
+    warning is printed and the user is skipped.
 
     Args:
         con: connection instance to connect to the DSP server
-        users: List of definitions of the users (JSON) to be created
-        groups: Dict with group definitions defined inside the actual ontology
-        project: Project the user(s) should be added to
+        users: "users" section of a parsed JSON project file
+        current_project_groups: groups defined in the current project (dict of the form {group name - group object}). These groups must exist on the DSP server.
+        current_project: "project" object of the current project (must exist on DSP server)
         verbose: Prints more information if set to True
 
     Returns:
-        None
+        True if all users could be created without any problems. False if a warning/error occurred.
     """
+    overall_success = True
     for user in users:
         username = user["username"]
 
-        # check if the user already exists, skip if so
-        maybe_user: Optional[User] = None
+        # skip the user if he already exists
         try:
-            maybe_user = User(con, email=user["email"]).read()
+            try_network_action(
+                object=User(con, email=user["email"]),
+                method="read",
+                error_message_on_failure=""
+            )
+            print(f"\tWARNING: User '{username}' already exists on the DSP server. Skipping...")
+            overall_success = False
+            continue
         except BaseError:
             pass
-        if maybe_user:
-            print(f"WARN User '{username}' already exists. Skipping...")
-            continue
 
-        sysadmin = False
```

Comment:
From here on, GitHub's diff renderling is a bit hard to understand. In PyCharm, it was more understandable. Basically, there are 4 things done inside the big loop `for user in users:`
 -  skip the user if he already exists
 -  if "groups" is provided, add user to the group(s)
 - if "projects" is provided, add user to the project(s)
 - create the user

I tried to simplify these four steps, to minimize the indentation depth, and to improve code understandability.

The diff might be hard to understand, but the code should be better understandable than before.